vm/src/obj/objtype.rs

```diff
@@ -213,13 +228,15 @@ pub fn init(ctx: &PyContext) {
                     type(name, bases, dict) -> a new type";
 
     extend_class!(&ctx, &ctx.types.type_type, {
+        "mro" => ctx.new_rustfunc(type_mro),
         "__call__" => ctx.new_rustfunc(type_call),
         "__dict__" =>
         PropertyBuilder::new(ctx)
                 .add_getter(type_dict)
                 .add_setter(type_dict_setter)
                 .create(),
-        "__new__" => ctx.new_rustfunc(type_new),
+        "__new__" => ctx.new_classmethod(type_new),
```

Comment:
Is this line nessessary? It seems `type` is a subclass of `object`

Reply:
It is, because `type` isn't in the mro of object subclasses.

```python
>>> class A: pass
... 
>>> A.mro()
[<class '__main__.A'>, <class 'object'>]
```

If there wasn't anything on `object`, `class_get_attr` wouldn't encounter `__new__` in the mro.

Wait no never mind, I see what you're saying. It only needs to be on `object` because `type` has `object` in *its* mro.